tests/static_analyzer/test_cpp_cdb_bear_generator.py

```
@@ -0,0 +1,373 @@
+            patch("static_analyzer.engine.adapters.cpp_cdb.bear_generator.shutil.which", side_effect=self._all_present),
+            patch("static_analyzer.engine.adapters.cpp_cdb.bear_generator.subprocess.run", side_effect=fake_run),
+        ):
+            with pytest.raises(RuntimeError, match=r"invalid compile_commands"):
+                BearGenerator(BuildSystemKind.MAKE).generate(tmp_path)
+
+        assert not cdb_path.exists()
+        assert not (cdb_dir / ".fingerprint").exists()
+
+
+class TestBearGeneratorAutotools:
+    """Autotools goes through 2–3 steps before bear runs."""
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)